src/rest/APIRequest.js

```diff
@@ -50,8 +50,8 @@ class APIRequest {
     let body;
     if (this.options.files?.length) {
       body = new FormData();
-      for (const file of this.options.files) {
-        if (file?.file) body.append(file.key ?? file.name, file.file, file.name);
+      for (const [index, file] of this.options.files.entries()) {
+        if (file?.file) body.append(`files[${index}]`, file.file, file.name);
```

Comment:
Has this been resolved?

Reply:
uploading stickers works for me, on this branch and on main

![image](https://user-images.githubusercontent.com/11778454/140630515-fefb4022-8a09-42fb-b7d3-82d84859840f.png)

Reply:
I'd still prefer that the `` `files[${index}]` `` logic is moved to the message code (which would just set that as file.key, for consistency) instead of the request code, but that isn't really necessary